CMakeLists.txt

```diff
@@ -30,6 +30,23 @@ project (sigc++)
 
 set( CMAKE_CXX_STANDARD 17 )
 
+# Turn on warnings for MSVC
+if (MSVC)
+    # Remove the CMake default of /W3 because when you add /W4, MSVC will complain
```

Comment:
I think this cmake PR will provide a better way to do this, so please link to it in the comment:
https://gitlab.kitware.com/cmake/cmake/merge_requests/3250

(Feel free to improve this commit and force push the whole PR branch.)

Reply:
That PR has been included in CMake 3.15 (the most recent major release), so I'll link to that. In fact - I can just add a version check in that `if()`, so if you'r eusing a new enough CMake, it's not done.